osbenchmark/utils/process.py

```diff
@@ -133,9 +133,12 @@ def run_subprocess_with_logging(command_line, header=None, level=logging.INFO, s
 
 def is_benchmark_process(p):
     cmdline = p.cmdline()
-    # On Linux, /proc/PID/status truncates the command name to 15 characters.
+    # On Linux, /proc/PID/status truncates the command name to 15 characters which matches both opensearch-benchmark
+    # and opensearch-benchmarkd. Hence, checking command line to exclude opensearch-benchmarkd process.
     return p.name() == "opensearch-benchmark" or \
-        p.name() == "opensearch-benc" or \
+        (p.name() == "opensearch-benc" and
+         len(cmdline) > 1 and
+         os.path.basename(cmdline[1]) != "opensearch-benchmarkd") or \
```

Comment:
Good catch!  Can this section simply be eliminated, since the process name reported by `/proc/PID/status` will be different in various circumstances anyway, and is not reliable.  For instance:
```
#!/usr/bin/python3:
Name:   opensearch-benc

#!/usr/bin/env python3: 
Name:   python3
```

Reply:
Makes sense. Raised https://github.com/opensearch-project/opensearch-benchmark/pull/553 to remove the truncated check.